.travis/build.sh

```diff
@@ -38,44 +38,61 @@ mvn spotbugs:check
 # Run testsuite with java 8 only
 if [ ${JAVA_MAJOR_VERSION} -eq 1 ] ; then
 
-  docker pull oryd/hydra:v1.8.5
+  arch=$(uname -m)
+
   docker pull quay.io/keycloak/keycloak:15.0.0
 
-  mvn test-compile spotbugs:check -e -V -B -f testsuite
-
-  set +e
-
-  clearDockerEnv
-  docker pull quay.io/strimzi/kafka:0.28.0-kafka-3.1.0
-  mvn -e -V -B clean install -f testsuite -Pkafka-3_1_0
-  EXIT=$?
-  exitIfError
-
-  clearDockerEnv
-  docker pull quay.io/strimzi/kafka:0.28.0-kafka-3.0.0
-  mvn -e -V -B clean install -f testsuite -Pkafka-3_0_0
-  EXIT=$?
-  exitIfError
-
-  clearDockerEnv
-  docker pull quay.io/strimzi/kafka:0.27.1-kafka-2.8.1
-  mvn -e -V -B clean install -f testsuite -Pkafka-2_8_1
-  EXIT=$?
-  exitIfError
-
-  clearDockerEnv
-  docker pull quay.io/strimzi/kafka:0.25.0-kafka-2.7.1
-  mvn -e -V -B clean install -f testsuite -Pkafka-2_7_1
-  EXIT=$?
-  exitIfError
-
-  clearDockerEnv
-  docker pull quay.io/strimzi/kafka:0.23.0-kafka-2.6.2
-  mvn -e -V -B clean install -f testsuite -Pkafka-2_6_2
-  EXIT=$?
-  exitIfError
-
-  set -e
+  if [ "$arch" == 's390x' ]; then
+    # Build s390x compatible hydra images
+    docker build --target hydra-import -t strimzi-oauth-testsuite/hydra-import:latest -f ./testsuite/docker/hydra-import/Dockerfile.s390x .
+    docker build --target oryd-hydra -t oryd/hydra:v1.8.5 -f ./testsuite/docker/hydra-import/Dockerfile.s390x .
+    mvn test-compile spotbugs:check -e -V -B -f testsuite
+    set +e
+    clearDockerEnv
+    docker pull quay.io/strimzi/kafka:0.29.0-kafka-3.1.0
```

Comment:
This image does not exist. So how come this is not failing?

Reply:
@scholzj - this was tested in my local dev environment where I had manually built the images.  For this reason testing is not yet enabled in `.travis.yml` for `s390x` platform.

Reply:
Ahh, ok. That makes sense. So I guess we should wait for the 0.29.0 release before merging this, so that we can enable it and test it? Hopefully, 0.29 will be out soon. We are waiting only for Kafka 3.1.1.

Reply:
During my testing I had made similar changes to test:
```
mvn test-compile spotbugs:check -e -V -B -f testsuite
set +e
clearDockerEnv
docker pull quay.io/strimzi/latest-kafka-3.1.0
docker tag quay.io/strimzi/latest-kafka-3.1.0 quay.io/strimzi/kafka:0.28.0-kafka-3.1.0
#docker pull quay.io/strimzi/kafka:0.28.0-kafka-3.1.0
mvn -e -V -B clean install -f testsuite -Pkafka-3_1_0
EXIT=$?
exitIfError
```

Reply:
Yes - let's wait until all images are available and initiate a test run.

Thanks @scholzj .

Reply:
0.29.0 is out now. So you should be able to try to update to it.

Reply:
Thanks @scholzj - I will test it shortly.

Reply:
@scholzj I tested it with 0.29.0 but ran into an issue where [testsuite/docker/hydra-import](https://github.com/strimzi/strimzi-kafka-oauth/blob/main/testsuite/docker/hydra-import/pom.xml) fails because docker plugin tries to build image using amd `Dockerfile`.  I am not sure if there is a way around it without bringing in `s390x` profile back as implemented earlier. 

Here is [TravisCI](https://app.travis-ci.com/github/linux-on-ibm-z/strimzi-kafka-oauth/jobs/571558217)  output which shows amd64 hydra image being built instead of `s390x`.

Let me know your thoughts.

Corresponding changes in `.travis/build.sh`:
```
  if [ "$arch" == 's390x' ]; then
    # Build s390x compatible hydra images
    export LD_LIBRARY_PATH=$LD_LIBRARY_PATH:/usr/lib/s390x-linux-gnu/jni
    docker build --target hydra-import -t strimzi-oauth-testsuite/hydra-import:latest -f ./testsuite/docker/hydra-import/Dockerfile.s390x .
    docker build --target oryd-hydra -t oryd/hydra:v1.8.5 -f ./testsuite/docker/hydra-import/Dockerfile.s390x .
    mvn test-compile spotbugs:check -e -V -B -f testsuite
    set +e
    clearDockerEnv
    docker pull quay.io/strimzi/kafka:0.29.0-kafka-3.1.1
    mvn -e -V -B clean install -f testsuite -Pcustom -Dkafka.docker.image=quay.io/strimzi/kafka:0.29.0-kafka-3.1.1
    EXIT=$?
    exitIfError
    set -e
  else
```

Reply:
I have no idea how the s390x environment works on Travis. Does the Docker engine run on s390x as well? If it can build amd64 hydra, can't it just run the amd64 hydra image in the first place? Then we would not need to build anything.

Reply:
My understanding is that TravisCI on `s390x` is LXD based and build jobs run within LXD container. Using amd64 hydra binary included in the hydra image will cause `exec format error` due to architectural differences. This particular problem only occurs for a specific testcase `hydra-import` of `docker` module testsuite.  Everything works (s390x specific) for `hydra-test` module.

Reply:
Maybe that needs to be moved into some shell script as well?

Reply:
I'll see if I can use plain `curl` in `hydra-import` so there is no need for installing 'hydra' executable in it. That would probably fix the issue. It would still use the 'alpine' image from Docker Hub though.

Reply:
Thanks @mstruk - I don't fully understand the code but `alpine` image should work on s390x as it is multi-arch enabled. I will be happy to test your changes on this platform.

If this works then changes in `.travis/build.sh` to build `oryd-hydra` image for `s390x` is all that is needed to address this issue.

Reply:
See https://github.com/linux-on-ibm-z/strimzi-kafka-oauth/pull/1

Reply:
@mstruk - looks good. 

Unfortunately, testcases continue to fail due to use of `Arquiliian Cube` extension which fails to interact with docker on `s390x`. I have tested a patch which compiles and installs s390x compatible `arquillian` jars in the local maven repo. This enables docker communication with `arquillian` on s390x.

For the purpose of this PR I think it is acceptable to have `hydra` issue addressed. I am away next week but will take this upon my return. Your changes above to remove hydra binary along with my changes to build `oryd:hydra` should take care of things.

Thanks!